transport/tls-record-frag/stream_dialer.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2023 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tlsrecordfrag
+
+import (
+	"context"
+	"errors"
+
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+)
+
+type tlsRecordFragDialer struct {
+	dialer     transport.StreamDialer
+	splitPoint int32
+}
+
+var _ transport.StreamDialer = (*tlsRecordFragDialer)(nil)
+
+// NewStreamDialer creates a [transport.StreamDialer] that splits the Client Hello Message
```

Comment:
We need to clarify the behavior.

```suggestion
// NewStreamDialer creates a [transport.StreamDialer] that splits the first TLS record if it's a handshake message. If the record has length N, the new records will have length prefixBytes and N - prefixBytes, plus their headers.
```

@jyyi1 let's make sure we agree on this behavior.